Libraries/Opc.Ua.Security.Certificates/Common/RsaUtils.cs

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 1996-2020 The OPC Foundation. All rights reserved.
+   The source code in this file is covered under a dual-license scenario:
+     - RCL: for OPC Foundation members in good-standing
+     - GPL V2: everybody else
+   RCL license terms accompanied with this source code. See http://opcfoundation.org/License/RCL/1.00/
+   GNU General Public License as published by the Free Software Foundation;
+   version 2 of the License are accompanied with this source code. See http://opcfoundation.org/License/GPLv2
+   This source code is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+*/
+
+using System;
+using System.Security.Cryptography;
+
+namespace Opc.Ua.Security.Certificates
+{
+    /// <summary>
+    /// Defines internal helper functions to implement RSA cryptography.
+    /// </summary>
+    internal static class RsaUtils
+    {
+        /// <summary>
+        /// Dispose RSA object only if not running on Mono runtime.
+        /// Workaround due to a Mono bug in the X509Certificate2 implementation of RSA.
+        /// see also: https://github.com/mono/mono/issues/6306
+        /// On Mono GetRSAPrivateKey/GetRSAPublickey returns a reference instead of a disposable object.
+        /// Calling Dispose on RSA makes the X509Certificate2 keys unusable on Mono.
+        /// Only call dispose when using .Net and .Net Core runtimes.
+        /// </summary>
+        /// <param name="rsa">RSA object returned by GetRSAPublicKey/GetRSAPrivateKey</param>
+        internal static void RSADispose(RSA rsa)
```

Comment:
Bug is closed since October 2018, is this special handling still necessary or is it safe to Dispose the RSA for mono?

Reply:
At this point I don't know. Would be great to get rid of it...

Reply:
lets make it a seperate work item